### PR TITLE
Adblock sticky banner - AB test

### PR DESCRIPTION
**Pull request by Calanthe**
This PR creates AB test which will test if users who use ad-blockers are willing to join membership.
There is a big banner (3 different variations - there are planned to be 2 more) which is located on top of the page and becomes sticky just like any other ad in top slot. 
Everyone inside AB test (who also use ad-blockers) should't see 'do not use ad-block' message shown at the bottom of the page. The AB test works on only tablet and desktop breakpoints.

Polly Toynbee:
tablet:
screen shot 2015-08-12 at 11 36 15

desktop (including leftCol and wide):
screen shot 2015-08-12 at 11 36 29

George Monbiot:
tablet:
screen shot 2015-08-12 at 11 51 25

desktop (including leftCol and wide):
screen shot 2015-08-12 at 11 51 14

Hugh Muir
tablet:
screen shot 2015-08-12 at 11 53 43

desktop (including leftCol and wide):
screen shot 2015-08-12 at 11 53 33

Banners are shown randomly.

### CODE REVIEW

**Comment by ScottPainterGNM** [comment]
Shouldn't the "5" and "£" be round the other way in the text?

**Comment by johnduffell** [comment]
Looks great!  I can't see any way to dismiss this banner - which is presumably by design.  However does it show once they become/already are a supporter?

Also a lot of users consider they are supporting the guardian in other ways (buying the paper or app subscriptions) and we should consider how to deal with them as it might offend them to suggest that they aren't already supporting the guardian.

File: static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            return detect.adblockInUse && detect.getBreakpoint() !== 'mobile';
+        };
+
+        this.variants = [{
+            id: 'variant',
+            test: function () {
+                var variations = [
+                    {
+                        supporterLink: 'https://membership.theguardian.com/about/supporter?INTCMP=ADBLOCK_BANNER_MONBIOT',
+                        quoteText: 'Become a Guardian Member and support independent journalism.',
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js]: ths punctuation isn't consistent - the other two messages don't have a full stop

File: static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        this.successMeasure = 'Ad-block users will be interested in becoming a Supporter';
+        this.audienceCriteria = 'Users who have ad-block installed and are not on mobile';
+        this.dataLinkNames = 'supporter message, read more';
+        this.idealOutcome = 'Users will sign up as a Supporter or will turn off the ad-block';
+
+        this.canRun = function () {
+            /**
+             * Adblock users who are not on mobile.
+             */
+            return detect.adblockInUse && detect.getBreakpoint() !== 'mobile';
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js]: the adblockInUse test is quite expensive at present especially on moobile, so I'd recommend swapping these two tests

File: static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js (hunk 1 above)
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js]: I just added full stops to be consistent.

**Comment by Calanthe** [comment]
@ScottPainterGNM Just fixed the order, thx for noticing it!

@johnduffell there is no option to dismiss this banner. And the issue with existing members - we are aware of that but currently there is no way to detect if user is a member. AFAIK membership team is working on it (or planing to do in the future).

**Comment by johnduffell** [comment]
yeah I'm looking forward to when we can integrate membership information into the site, but I know some users are easily annoyed if they feel they have legitimate reason to use an ad blocker.  and they don't care whether there's even a way for us to know - people even complained that they saw the old message when they buy the print edition every day, which we have no way of knowing!

having said that, most users are probably fine with it and if it's already considered and we're happy enough with the wording then go ahead :+1:

File: static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js (hunk 1 above)
Comment (Jholder112233) [inline static/src/javascripts/projects/common/modules/experiments/tests/adblock-sticky-banner.js]: Minor point, but house style would be to drop the full stops at the end.

Also Polly's "If you read the Guardian. Join the Guardian." should be "If you read the Guardian, join the Guardian"

**Pull request merged**